### DISCUSSION TITLE
How to properly make Grid and DataGrid scrollable?

### DISCUSSION THREAD

**Discussion by Gigas002** [AvaloniaUI/Avalonia#7458] [Q&A] [answered]
The question's the same as title. I searched for `scroll` in [UserControl](http://reference.avaloniaui.net/api/Avalonia.Controls/UserControl/), [Grid](http://reference.avaloniaui.net/api/Avalonia.Controls/Grid/) and [DataGrid](http://reference.avaloniaui.net/api/Avalonia.Controls/DataGrid/) API and it seems like only `DataGrid` has an ability to be scrollable with `VerticalScrollBarVisibility`/`HorizontalScrollBarVisibility` properties, though this works time to time (when I resize window to full size and then resize it back the `DataGrid` is no longer scrollable).
Is there a better way to make `DataGrid` scrollable? And is there any way to make `Grid` scrollable?

**Comment by jp2masa** [comment, marked as answer]
For `DataGrid` it may be a bug. For `Grid` or any panel you can usually just wrap it inside a `ScrollViewer`, e.g.:

```xaml
<ScrollViewer>
    <Grid>
    </Grid>
</ScrollViewer>
```

For `DataGrid` I believe it already has a `ScrollViewer` in the template, in the body, so that the headers are always visible.

**Reply by Gigas002** [reply]
Thank you for your answer, that's pretty much the solution I was looking for. Though I think it's designed kinda weird, to make so simple thing with wrapping around a new control instead of a property
